src/test/ui/try-trait/bad-interconversion.stderr

```diff
@@ -40,12 +40,12 @@ LL | | }
    = help: the trait `FromResidual<ControlFlow<{integer}, Infallible>>` is not implemented for `Result<u64, String>`
    = note: required by `from_residual`
 
-error[E0277]: the `?` operator can only be used on `Option`s in a function that returns `Option`
+error[E0277]: the `?` operator can only be used on `Option`s, not `Result`s, in a function that returns `Option`
   --> $DIR/bad-interconversion.rs:22:22
    |
 LL | / fn result_to_option() -> Option<u16> {
 LL | |     Some(Err("hello")?)
-   | |                      ^ this `?` produces `Result<Infallible, &str>`, which is incompatible with `Option<u16>`
+   | |                      ^ use `.ok()?` if you want to discard the `Result<Infallible, &str>` error information
```

Comment:
I wonder if it would be clearer if we mentioned something like `if you want to discard the Result::Err(&str)` or something along those lines 🤔

Reply:
Yeah, the residual being "coloured" with the `Try` type does make it harder to put in this error message nicely.  I guess it would be less bad with just `Result<!, &str>`, but `Infallible` is long enough to be distracting. 🤷

Reply:
I'm more worried about confusing newcomers than anything else, anyone that already knows about `Result`, `Option` and `.ok()` wouldn't be confused regardless of wording, but someone who doesn't yet have a good grasp of what an `enum` even or that hasn't internalized `Result::Err` is the error information might be helped with things being more spelled out. Either way, I'm ok with the current wording.

Reply:
I *have* been looking at the quadratic growth of the on-unimplemented attribute here and been wondering if it should turn into custom code for it at some point.  If at some point in the future you get inspired with a great idea for how these error should look, feel free to assign me an issue.  (Or, of course, just do it yourself if that's easier that answering my inevitable noob questions 😅)